### PR TITLE
Update dependencies and remove redux

### PR DESCRIPTION
**Pull request by aulneau**
> Try out this version of the Stacks Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/836108507).

This PR removed the little bit of redux we used in favor of recoil, and also updates all of our dependencies to their latest versions.

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]
Nice refactor! I was just reading up this weekend on Recoil bc I haven't used. I was wondering why we had both Redux and Recoil in the repo, so this is helpful!

File: src/common/api/accounts.ts [hunk 1]
```diff
@@ -7,38 +7,38 @@ import type {
...
-): Promise<AddressBalanceResponse> => {
-  const path = `/address/${principal}/balances`;
-  return fetchFromSidecar(apiServer)(path);
-};
+export const fetchBalances =
+  (apiServer: string) =>
+  (principal: string): Promise<AddressBalanceResponse> => {
+    const path = `/address/${principal}/balances`;
+    return fetchFromSidecar(apiServer)(path);
+  };
```
Comment (kyranjamie) [inline src/common/api/accounts.ts:15]: What's going on here? This isn't how prettier would format this?

**Review by kyranjamie — CHANGES_REQUESTED** [review]
Code changes look good but something's up with your prettier set up
```
kyranjamie in ~/dev/hiro/stacks-wallet-web                                                                                                                                                                                                                                                                              ±[fix/state-cleanup]
❯ yarn prettier --check "src/**/*.{ts,tsx}"
yarn run v1.22.10
warning package.json: "dependencies" has dependency "schema-inspector" with range "2.0.1" that collides with a dependency in "devDependencies" of the same name with version "2.0.0"
$ /Users/kyranjamie/dev/hiro/stacks-wallet-web/node_modules/.bin/prettier --check 'src/**/*.{ts,tsx}'
Checking formatting...
src/common/api/accounts.ts
src/common/api/fetch.ts
src/common/api/transactions.ts
src/common/hooks/callbacks/use-decode-request-callback.ts
src/common/hooks/callbacks/use-network-switch-callback.ts
src/common/hooks/callbacks/use-save-auth-request-callback.ts
src/common/hooks/use-onboarding-state.ts
src/common/hooks/use-tx-state.ts
src/common/hooks/use-vault-messenger.ts
src/common/hooks/use-wallet.ts
src/common/onboarding-data.ts
src/common/stacks-utils.ts
src/common/transaction-utils.ts
src/components/color-modes.tsx
src/components/drawer/accounts/index.tsx
src/components/drawer/accounts/switch-accounts.tsx
src/components/popup/settings-popover.tsx
src/pages/install/sign-in.tsx
src/pages/transaction/edit-post-conditions.tsx
src/store/index.ts
src/store/transaction.ts
src/store/transfers.ts
Code style issues found in the above file(s). Forgot to run Prettier?
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: src/common/api/accounts.ts (hunk 1 above)
Comment (aulneau) [reply src/common/api/accounts.ts:15]: https://prettier.io/blog/2021/05/09/2.3.0.html

Comment (kyranjamie) [reply src/common/api/accounts.ts:15]: cooo00ooOOool. Would be great to get this update added via the prettier-config though

**Review by kyranjamie — APPROVED** [review]

File: src/common/api/accounts.ts (hunk 1 above)
Comment (aulneau) [reply src/common/api/accounts.ts:15]: @kyranjamie what changes are required?

**Pull request merged**